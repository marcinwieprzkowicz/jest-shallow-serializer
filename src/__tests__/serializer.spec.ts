const { serialize } = require("../serializer");

describe("serializer", () => {
  it("replaces node with `data-shallow-name` attribute", () => {
    const element = document.createElement("div");
    element.setAttribute("data-shallow-name", "MyComponentName");
    element.setAttribute("data-shallow-props", "{}");

    const output = serialize(element, null, null, 0);
    expect(output).toEqual("<MyComponentName />");
  });

  it("renders component properties", () => {
    const props = {
      obj: {
        bool: true,
        objectEmpty: {},
        arrayEmpty: [],
        arrayNumbers: [1, 2, 3],
        arrayStrings: ["a", "b", "c"],
        methodNormal: "[Function]",
        nestedObject: {
          methodMocked: "[MockFunction]",
          stringProperty: "dolor sit amet",
        },
        stringProperty: "lorem ipsum",
        numberProperty: 5,
      },
      arrayObjects: [
        {
          firstName: "John",
          lastName: "Doe",
        },
        {
          firstName: "Jan",
          lastName: "Kowalski",
        },
      ],
      bool: false,
      objectEmpty: {},
      arrayEmpty: [],
      arrayNumbers: [1, 2, 3],
      arrayStrings: ["a", "b", "c"],
      stringProperty: "lorem ipsum",
      numberProperty: 6,
    };
    const element = document.createElement("div");
    element.setAttribute("data-shallow-name", "MyComponentName");
    element.setAttribute("data-shallow-props", JSON.stringify(props));

    const output = serialize(element, null, null, 0);
    expect(output).toMatchSnapshot();
  });
});
