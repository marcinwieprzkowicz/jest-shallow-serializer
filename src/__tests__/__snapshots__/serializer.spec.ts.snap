// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serializer renders component properties 1`] = `
"<MyComponentName
  obj={
    {
      bool: true,
      objectEmpty: {},
      arrayEmpty: [],
      arrayNumbers: [
        1,
        2,
        3,
      ],
      arrayStrings: [
        "a",
        "b",
        "c",
      ],
      methodNormal: [Function],
      nestedObject: {
        methodMocked: [MockFunction],
        stringProperty: "dolor sit amet",
      },
      stringProperty: "lorem ipsum",
      numberProperty: 5,
    }
  }
  arrayObjects={
    [
      {
        firstName: "John",
        lastName: "Doe",
      },
      {
        firstName: "Jan",
        lastName: "Kowalski",
      },
    ]
  }
  bool={false}
  objectEmpty={{}}
  arrayEmpty={[]}
  arrayNumbers={
    [
      1,
      2,
      3,
    ]
  }
  arrayStrings={
    [
      "a",
      "b",
      "c",
    ]
  }
  stringProperty="lorem ipsum"
  numberProperty={6}
/>"
`;
